src/lib/lib.jl

```diff
@@ -1,5 +1,11 @@
 using Base: RefValue
 
+if VERSION > v"1.7.0-DEV.204"
+  using Base: ismutabletype
+else
+  ismutabletype(::Type{T}) where T = T.mutable
```

Comment:
```suggestion
  function ismutabletype(@nospecialize(t::Type))
    t = Base.unwrap_unionall(t)
    return isa(t, DataType) && t.mutable
  end
```

Reply:
Maybe better? It used to literally call `T.mutable` so that's all I was trying to preserve.

Reply:
Just feel it better to keep consistent with the upstream implementation if we want to use the name `ismutabletype` 😆

Reply:
Yes am using that on new versions; CI informed me that it's a recent addition. 

I guess if there any difference in specialisation performance, we don't care as it's only used inside `@generated`, in fact would prefer less specialisation always.

Reply:
Let's call CI sufficiently happy